test-locally.sh

```
@@ -251,39 +263,70 @@ cleanup() {
-  { wait 2>/dev/null || true; } 2>/dev/null
-  
-  # Check if validators are actually stopped
-  if ! pgrep -f "solana-test-validator" >/dev/null 2>&1 && ! pgrep -f "ephemeral-validator" >/dev/null 2>&1; then
-    echo "✓ Stopped"
+  # Only stop validators if we started them
+  if [ "$SOLANA_STARTED_BY_US" = true ] || [ "$EPHEMERAL_STARTED_BY_US" = true ]; then
+    echo -n "Stopping validators... "
+    
+    # Kill by PID if available and we started them
+    if [ "$SOLANA_STARTED_BY_US" = true ] && [ -n "$SOLANA_PID" ]; then
+      kill -TERM $SOLANA_PID 2>/dev/null || true
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)